src/js/core/methods/CardanoGetAddress.js

```diff
@@ -76,6 +78,7 @@ export default class CardanoGetAddress extends AbstractMethod {
                 address: batch.address,
                 protocol_magic: batch.protocolMagic,
                 network_id: batch.networkId,
+                derivation_type: batch.derivation_type || CardanoDerivationType.ICARUS_TREZOR,
```

Comment:
I think this is not correct - `CardanoDerivationType.LEDGER` has value `0`, so it gets replaced by `CardanoDerivationType.ICARUS_TREZOR`. I tested it (with the mnemonic from [firmware derivation tests](https://github.com/vacuumlabs/trezor-firmware/blob/master/common/tests/fixtures/cardano/get_base_address.derivations.json)) and indeed `LEDGER` gives the same signature as `ICARUS_TREZOR` (which is different from the `ICARUS` one).

This line is present multiple times below.

Reply:
!!! great observation. thanks. I will fix that and also add some tests to cover this.